lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -176,6 +176,57 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
     end
   end
 
+  @doc """
+
+  """
+  @spec fetch_identifier(
+          charlist(),
+          {:module, module()} | {:function | :type, name :: atom(), arity :: pos_integer()}
+        ) ::
+          {:ok, cell_id :: String.t(), line :: pos_integer()} | :error
+  def fetch_identifier(path, identifier)
+
+  def fetch_identifier(path, {:module, module}) do
+    with {:ok, {:raw_abstract_v1, annotations}} <- beam_lib_chunks(path, :abstract_code) do
+      {:attribute, anno, :module, ^module} =
+        Enum.find(annotations, &match?({:attribute, _, :module, _}, &1))
+
+      {:ok, cell_id_from_module(path), :erl_anno.line(anno)}
+    end
+  end
+
+  def fetch_identifier(path, {:function, name, arity}) do
+    with {:ok, {:debug_info_v1, _, {_, metadata, _}}} <- beam_lib_chunks(path, :debug_info) do
+      {_, :def, [line: line], _} =
+        Enum.find(metadata.definitions, &match?({{^name, ^arity}, :def, [line: _line], _}, &1))
+
+      {:ok, cell_id_from_module(path), line}
+    end
+  end
+
+  def fetch_identifier(path, {:type, name, arity}) do
+    bin = File.read!(path)
+
+    with {:ok, types} <- Code.Typespec.fetch_types(bin) do
+      {:ok, line} =
```

Comment:
There is no guarantee you will find the given name/arity, right?

So I think you want this inside `with`:

    {:ok, types} <- Enum.find_value(types, :error, fn ....